### PR TITLE
El 20724 micromter refactor

### LINKED ISSUE #20724: Refactor internal exporter metrics to use Micrometer
**Is your feature request related to a problem? Please describe.**

I would like to remove the explicit dependency on prometheus as the backend for the metrics. Enabling in the future a change to any monitoring system just by simply changing the meter registry metrics are registered to.

**Describe the solution you'd like**
Currently metrics in the `OpensearchExporter` and `MetricsExporter` use the prometheus client library. Convert the metrics exported to be done using the Micrometer core library. 

Some metric data structures work identically such as the `Counter` however some such as a `Gauge` require different client code to interact with (even if they represent the same underlying data). There is an example here #20590 where the elastic search exporter metrics was refactored to use the micrometer core library.

### PR DESCRIPTION
**Pull request by EuroLew**
## Description

Refactor metricsExporter and the Opensearch exporter to use micrometer for metrics instead of prometheus.

## Related issues

closes #20724

### CODE REVIEW

File: zeebe/exporters/opensearch-exporter/src/main/java/io/camunda/zeebe/exporter/opensearch/OpensearchMetrics.java
Comment (npepinpe) [inline zeebe/exporters/opensearch-exporter/src/main/java/io/camunda/zeebe/exporter/opensearch/OpensearchMetrics.java:35]: :question: How does this work? I understand last time that we had to pass a runnable or something to record a method, but now this works with the try-with-resources pattern?

**Review by npepinpe — CHANGES_REQUESTED** [review]
:rocket: 

:x: One thing, I'm not sure why we rebuild the metrics all the time. It's creating intermediate objects unnecessarily every time right? Can't we create them once at initialization and then whenever the `measureSomething` is called, we just use the instance (e.g. a timer)?

File: zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/metrics/ExecutionLatencyMetrics.java
Comment (EuroLew) [reply zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/metrics/ExecutionLatencyMetrics.java:58]: There is no way to tag a meter dynamically and ALSO provide a description and set the sla buckets. There is only the for example `meterRegistry.timer(name, <tags>)` which will create a timer in the registry, to get a description and set the buckets it will require setting the meter registry config such that these get attached when a specific timer name gets created.

The reason this is done here differently than the `ElasticSearchMetrics` where its created in the constructor, is that the ELS metrics does not need to dynamically tag the partitionId, there it is provided in the constructor and all metrics recorded against that instance will use the partitionId in the constructor.

However here the functions which observe metrics also take a partitionId which in micrometer means a new object (only if the timer doesn't exist the documentation says `The returned timer will be unique for each registry, but each registry is guaranteed to only create one timer for the same combination of name and tags.`

Therefore my understanding is they are only created the first time a set of metric name and tags is observed.

File: zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/metrics/MetricsExporter.java
Comment (EuroLew) [reply zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/metrics/MetricsExporter.java:63]: fixed. Passed null, should be fine as it is then overwritten in the configure.

File: zeebe/exporters/opensearch-exporter/src/main/java/io/camunda/zeebe/exporter/opensearch/OpensearchMetrics.java
Comment (EuroLew) [reply zeebe/exporters/opensearch-exporter/src/main/java/io/camunda/zeebe/exporter/opensearch/OpensearchMetrics.java:35]: Yes it is now used in the try-with-resources pattern, `ResourceSample` implements `AutoCloseable`

File: zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/metrics/ExecutionLatencyMetrics.java
Comment (npepinpe) [reply zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/metrics/ExecutionLatencyMetrics.java:58]: I see. This is a holdover from the past where the exporter did not know the partition ID from the get go, and read it from the record during exporting. This could actually be refactored now, since we already know the partition ID in the configure method of the exporter, so we could create the metrics class at that point and have no dynamic tags.

Considering it's a small change, I would do it in this PR, but I'm fine with doing it as a follow up.

**Review by npepinpe — APPROVED** [review]
:rocket: Some comments, but no need for another review imo. Please address them before merging.

File: zeebe/exporters/opensearch-exporter/src/main/java/io/camunda/zeebe/exporter/opensearch/OpensearchMetrics.java
Comment (EuroLew) [reply zeebe/exporters/opensearch-exporter/src/main/java/io/camunda/zeebe/exporter/opensearch/OpensearchMetrics.java:35]: done

File: zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/metrics/MetricsExporter.java
Comment (EuroLew) [reply zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/metrics/MetricsExporter.java:80]: done

File: zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/metrics/ExecutionLatencyMetrics.java
Comment (EuroLew) [reply zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/metrics/ExecutionLatencyMetrics.java:58]: > ❌ Why not create these objects in the constructor to avoid having to recreate them here every time?

Also not sure what is meant my this, they are not recreated every time only the first time during creation. Am I misunderstanding something?

**Comment by npepinpe** [comment]
Should we merge?

**Pull request merged**
